Assets/Scripts/Gameplay/Action/BaseTypes/ActionFactory.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using Object = UnityEngine.Object;
+
+namespace Unity.Multiplayer.Samples.BossRoom.Actions
+{
+    public static class ActionFactory
+    {
+        /// <summary>
+        /// Factory method that creates Actions from their request data.
+        /// </summary>
+        /// <param name="data">the data to instantiate this skill from. </param>
+        /// <returns>the newly created action. </returns>
+        public static Action CreateActionFromData(ref ActionRequestData data)
+        {
+            var actionPrototype = GameDataSource.Instance.GetActionPrototypeByID(data.ActionID);
+
+            var ret = Object.Instantiate(actionPrototype);
+            ret.Initialize(ref data);
+            ret.RuntimePrototypeReference = actionPrototype;
+            ret.ActionID = actionPrototype.ActionID;
+            return ret;
+        }
+
+        //todo pool acitons by type and purge when asked to
```

Comment:
any jira tasks for those todos? just so they are not lost

Reply:
Pooling I implemented, and the DI stuff is irrelevant